File: LibraryAPI/src/schemas/TakingBook/taking_book_create.py
from pydantic import BaseModel
from datetime import date


class TakingBookCreate(BaseModel):
    book_id: int
    user_id: int
    employee_id: int
    taking_date: date  # Обязательное поле
    is_returned: bool = False  # По умолчанию книга не возвращена
    return_date: date | None = None  # Может быть пустым, если не возвращена

File: LibrarryApp/LibrarryDesktop/Models/ApiRequestModels/TakingBookCreate.cs
using Newtonsoft.Json;

namespace LibraryDesktop.Models.ApiRequestModels
{
    public class TakingBookCreate
    {
        [JsonProperty("book_id")]
        public int BookId { get; set; }
        [JsonProperty("user_id")]
        public int UserId { get; set; }
        [JsonProperty("employee_id")]
        public int EmployeeId { get; set; }
        [JsonProperty("taking_date")]
        public DateTime TakingDate { get; set; }
        [JsonProperty("is_returned")]
        public bool IsReturned { get; set; }
        [JsonProperty("return_date")]
        public DateTime? ReturnDate { get; set; }
    }
}